authorization-server/src/modules/auth/roles-validation.service.ts

```diff
@@ -0,0 +1,31 @@
+import { Injectable } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { LoggerService } from '../logger';
+import { IDidAccessTokenPayload } from './types';
+
+@Injectable()
+export class RolesValidationService {
+  constructor(
+    private readonly configService: ConfigService,
+    private readonly logger: LoggerService,
+  ) {
+    this.logger.setContext(RolesValidationService.name);
+  }
+
+  public async didAccessTokenRolesAreValid(
+    verifiedRoles: IDidAccessTokenPayload['verifiedRoles'],
+  ): Promise<boolean> {
+    const acceptedRoles = this.configService
+      .get('ACCEPTED_ROLES', '')
+      .split(',');
```

Comment:
Given that is it expected that `acceptedRoles` always has an entry, maybe it's worth throwing an error here or at a minimum logging an error and returning?
```suggestion
    const acceptedRoles = this.configService
      .get('ACCEPTED_ROLES', '')
      .split(',');
    if (acceptedRoles?.length < 1) {
      this.logger.error(`acceptedRoles should contain at least one entry`);
      return false;
    }
```

Reply:
Yeah, I am adding an error message here. This is an unexpected state and it's better to be more verbose here. Good idea.